src/meta/src/manager/catalog/mod.rs

```diff
@@ -2221,6 +2221,15 @@ where
         }
         tables
     }
+
+    pub async fn get_created_table_ids(&self) -> Vec<u32> {
```

Comment:
This method only list the id in the `Table` catalog and index is missing since it is in `Index` catalog. To make sure the list of created table ids is consistent with the actual created state table ids. I suggest we add the following method in `FragmentManager` and use it for compaction group split:
```
pub async fn list_created_state_table_ids(&self) -> Vec<u32> {
        self.core
            .read()
            .await
            .table_fragments
            .values()
            .filter(|f| f.is_created())
            .map(|f| f.all_table_ids())
            .collect_vec()
    }
```

Reply:
All index table have existed in `tables`.